sklearn/tree/_criterion.pyx

```diff
@@ -230,10 +261,19 @@ cdef class ClassificationCriterion(Criterion):
         self.start = 0
         self.pos = 0
         self.end = 0
+        self.missing_direction = 2  # Ignore the missing values by default
 
         self.n_outputs = n_outputs
+
+        self.n_missing = 0
+        self.weighted_n_missing = 0
+
+        self.n_nonmissing = 0
+        self.weighted_n_nonmissing = 0
```

Comment:
Also I've modified this PR a lot since updating them. I'll push them and ping you :) I think it might convince you on why `weighted_n_available` is needed instead of computing them on the fly. :)